src/Components/InfiniteScrollPreviousPage.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from "react";
+import { compose, pure, setDisplayName } from "recompose";
+import { IModel, IQueryBuilder, IQueryManager } from "redux-data-service";
+import { Omit } from "redux-data-service/dist/Omit";
+
+import { withModelQuery } from "../WithModelQuery";
+import { omitProps } from "../Helpers";
+
+export interface IInfiniteScrollPreviousPageProps {
+  query: IQueryBuilder;
+  modelComponent: React.ComponentType<{ model: IModel<any> } & any>;
+  modelComponentProps?: any;
+}
+
+interface IInfiniteScrollPreviousPageInternalProps extends Omit<IInfiniteScrollPreviousPageProps, "query"> {
+  query: IQueryManager<any>;
+}
+
+export const InfiniteScrollPreviousPage = compose<IInfiniteScrollPreviousPageInternalProps, IInfiniteScrollPreviousPageProps>(
+  setDisplayName("InfiniteScrollPreviousPage"),
+  pure,
+  withModelQuery({ isLoading: ({ query }) => query.response == null }),
+)(({
+  query,
+  modelComponent: ModelComponent,
+  modelComponentProps
+}) => (
+  <React.Fragment key={`page-${query.response.currentPage}`}>
+    {query.hasPreviousPage() && (
+      <InfiniteScrollPreviousPage
```

Comment:
I'm confused about why this is calling itself (?)

Reply:
This is an existing component that was simply renamed, moved to its own file and cleaned up. It recursively calls itself to render all previous pages, used by the "infinite scroll" mechanism (when `disableVirtualScrolling` is set to `true` on `InfiniteScroll`). For example, if we're on page 4, this will render page 1, page 2, page 3, page 4.

Reply:
Makes sense.